### PR TITLE
Gherkin for export of application in topology

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-2148
Story: https://issues.redhat.com/browse/ODC-6280

Acceptance criteria:

- Users can access an Export CTA from Topology to export resources in the selected namespace. Note that users will not be able to pick and choose which resources will be exported.

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

### CODE REVIEW

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
+@odc-2148
+Feature: Export of application
+              As a user, I have an unmanaged application which I want to export. I'd like to be able to later add that code to git or some shared location so that I can share with others, or import into a new cluster or same cluster but different project, or be able to apply updates to an existing application.
+
+
+
+    #The operator mentioned below is not yet available on operator hub
+        Background:
+            Given user has installed Gitops primer Operator
+              And user has created or selected namespace "aut-export-application"
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:10]:
```suggestion
            Given user has installed Gitops primer Operator
              And user is at developer perspective
              And user has created or selected namespace "aut-export-application"
```

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+        @smoke @to-do
+        Scenario: Export application button in topology: EA-02-TC01
+            Given user is at Developer perspective
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:15]: Move it to BackGround section and remove it from here

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              And user has created a deployment workload
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:16]: Provide the test data "<workloadName>"

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+             When user goes to Topology
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:17]:
```suggestion
             When user navigates to Topology
```

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+              And user clicks on Export Application button
+             Then user can see a toast message saying "Export of resources in aut-export-application has started."
+              And user can see primer job created in topology
+
+
+        @regression @to-do
+        Scenario: Export Application modal when export application is in progress: EA-02-TC02
+            Given user is at Topology
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:25]:
```suggestion
            Given user is at Topology page
```

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+             When user clicks on Export Application button
+              And user clicks on Export Application button again
+             Then user can see a Modal saying "Export Application in primer app is in progress" with Ok button
+
+
+        @regression @to-do
+        Scenario: Export application button in empty topology view: EA-02-TC03
+            Given user is at Developer perspective
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:33]: Remove this line and move it to BackGround section

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+              And user has created or selected namespace "aut-export-applicatio-temp"
+             When user goes to Topology
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:35]:
```suggestion
             When user navigates to Topology page
```

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+             Then user can see Export Application button disabled
+
+
+        @regression @manual
+        Scenario: Export Application is created: EA-02-TC04
+            Given user is at Topology
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:41]:
```suggestion
            Given user is at Topology page
```

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+              And Export Application is completed
+             Then user can see a toast message on export completion along with a link to download
+              And user can see a deployment workload for primer is created
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:44]: In Given condition,
'User created deployment workload "<>"' is required and share that workload name here in this step for more clarity

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+
+        @regression @manual
+        Scenario: Download the exported application: EA-02-TC05
+            Given user is at Topology
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:49]:
```suggestion
            Given user is at Topology page
```

**Review by makambalaji — CHANGES_REQUESTED** [review]
Do we need "export-application" folder in Topology plugin folder?

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 9 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature:44]: No, this is a workload when the primer Job is finished the primer deployment is created

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Added

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 5 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 7 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: v

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 8 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

File: frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature (hunk 10 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/export-application/export-of-applicaton.feature]: Updated

**Comment by sanketpathak** [comment]
Yes, as the feature code is present in the Topology plugin so we need the export application tests in Topology at this moment

**Comment by invincibleJai** [comment]
/approve 

cc @makambalaji

**Pull request merged**